### PR TITLE
Handle Spotify authentication better

### PR DESCRIPTION
**Pull request by albe2669**
This PR makes the following changes to vastly improve the authentication flow to the Spotify servers:
- Attempt to read the token from a configured file
- When getting a token from Spotify write it to said faile
- Wrap all API calls in a callSpotify tool that checks if the token needs to be refreshed

Commits:
- **Create player endpoint to test**
- **Read and write token from file\n\nCo-authored-by: Slug-Boi <theis.p.holm@gmail.com>**
- **Add token to gitignore**
- **Better auth flow**

### CODE REVIEW

File: lib/spotify/spotify.go [hunk 1]
```diff
@@ -24,9 +27,31 @@ func (s *Spotify) SetupRoutes(e *echo.Echo) {
...
+
+	s.auth.authenticate()
+}
+
+// Don't question it: https://groups.google.com/g/golang-nuts/c/y9IvZgiNowk
+func callSpotify[R *Q, Q any](ctx context.Context, spot *Spotify, f func(ctx context.Context, opts ...spotifyLib.RequestOption) (R, error), opts ...spotifyLib.RequestOption) (R, error) {
+	var err error
+
+	if spot.Client == nil {
+		// TODO: return a custom error type
```
Comment (Slug-Boi) [inline lib/spotify/spotify.go:42]: Isn't this already done?
```suggestion
```

**Review by Slug-Boi — APPROVED** [review]
Approved just have one small suggestion you can look at

File: lib/spotify/spotify.go (hunk 1 above)
Comment (albe2669) [reply lib/spotify/spotify.go:41]: It kind of was, but i meant an actual type, i've fixed it in the newest commit. Thank you for the reminder.

**Review by Slug-Boi — APPROVED** [review]
Ah that makes more sense. Looks like a good error good job!

**Pull request merged**